cloudinit/sources/DataSourceAzure.py

```diff
@@ -1338,76 +1338,101 @@ def load_azure_ds_dir(source_dir):
-                            dev_config['dhcp6-overrides'] = dhcp_override
-                    for addr in addresses[1:]:
-                        # Append static address config for ip > 1
-                        netPrefix = intf[addr_type]['subnet'][0].get(
-                            'prefix', default_prefix)
-                        privateIp = addr['privateIpAddress']
-                        if not dev_config.get('addresses'):
-                            dev_config['addresses'] = []
-                        dev_config['addresses'].append(
-                            '{ip}/{prefix}'.format(
-                                ip=privateIp, prefix=netPrefix))
-                if dev_config:
-                    mac = ':'.join(re.findall(r'..', intf['macAddress']))
-                    dev_config.update({
-                        'match': {'macaddress': mac.lower()},
-                        'set-name': nicname
-                    })
-                    # With netvsc, we can get two interfaces that
-                    # share the same MAC, so we need to make sure
-                    # our match condition also contains the driver
-                    driver = device_driver(nicname)
-                    if driver and driver == 'hv_netvsc':
-                        dev_config['match']['driver'] = driver
-                    netconfig['ethernets'][nicname] = dev_config
-            evt.description = "network config from imds"
-        else:
-            blacklist = ['mlx4_core']
-            LOG.debug('Azure: generating fallback configuration')
-            # generate a network config, blacklist picking mlx4_core devs
-            netconfig = net.generate_fallback_config(
-                blacklist_drivers=blacklist, config_driver=True)
-            evt.description = "network config from fallback"
-        return netconfig
+    if imds_metadata != sources.UNSET and imds_metadata:
+        try:
+            return _generate_network_config_from_imds_metadata(
+                imds_metadata)
+        except Exception as e:
+            msg = 'Failed generating network config from IMDS metadata: %s' % e
+            report_diagnostic_event(msg)
+            LOG.warning(msg)
+    try:
+        return _generate_network_config_from_fallback_config()
+    except Exception as e:
+        msg = 'Failed generating fallback network config: %s' % e
+        report_diagnostic_event(msg)
+        LOG.warning(msg)
+    return {}
+
+
+@azure_ds_telemetry_reporter
+def _generate_network_config_from_imds_metadata(imds_metadata) -> dict:
+    """Convert imds_metadata dictionary to network v2 configuration.
+    Parses network configuration from imds metadata.
+
+    @param: imds_metadata: Dict of content read from IMDS network service.
+    @return: Dictionary containing network version 2 standard configuration.
+    """
+    netconfig = {'version': 2, 'ethernets': {}}
+    report_diagnostic_event(
+        'Azure: generating network configuration from IMDS')
```

Comment:
I would drop the "Azure: " prefix, it's redundant and the rest of the code from this file doesn't use it. In fact I don't think you need this diagnostic event, it does not add any additional information. The telemetry reporter will generate telemetry to indicate that the code is invoking this function, which means it's generating configuration from IMDS